rllib/algorithms/dreamerv3/tf/models/disagree_networks.py

```diff
@@ -0,0 +1,100 @@
+"""
+[1] Mastering Diverse Domains through World Models - 2023
+D. Hafner, J. Pasukonis, J. Ba, T. Lillicrap
+https://arxiv.org/pdf/2301.04104v1.pdf
+"""
+
+import tensorflow as tf
+
+from ray.rllib.algorithms.dreamerv3.tf.models.components.mlp import MLP
+from ray.rllib.algorithms.dreamerv3.tf.models.components.representation_layer import (
+    RepresentationLayer,
+)
+
+
+class DisagreeNetworks(tf.keras.Model):
+    """Predict the RSSM's z^(t+1), given h(t), z^(t), and a(t).
+
+    Disagreement (stddev) between the N networks in this model on what the next z^ would
+    be are used to produce intrinsic rewards for enhanced, curiosity-based exploration.
+
+    TODO
+    """
+
+    def __init__(self, *, num_networks, model_dimension, intrinsic_rewards_scale):
+        super().__init__(name="disagree_networks")
+
+        self.model_dimension = model_dimension
+        self.num_networks = num_networks
+        self.intrinsic_rewards_scale = intrinsic_rewards_scale
+
+        self.mlps = []
+        self.representation_layers = []
+
+        for _ in range(self.num_networks):
+            self.mlps.append(
+                MLP(
+                    model_dimension=self.model_dimension,
+                    output_layer_size=None,
+                    trainable=True,
+                )
+            )
+            self.representation_layers.append(
+                RepresentationLayer(
+                    model_dimension=self.model_dimension, name="disagree"
+                )
+            )
+
+        # Optimizer.
+        self.optimizer = tf.keras.optimizers.Adam(learning_rate=1e-4, epsilon=1e-5)
+
+    # @tf.functionction
+    def call(self, inputs, z, a, training=None):
+        return self.forward_train(a=a, h=inputs, z=z)
+
+    # @tf.functionction
```

Comment:
remove?

Reply:
Ah, sorry, yes, these should all be removed, I guess. eager tracing is now entirely done via RLlib/RLModules.

Reply:
done